### PR TITLE
Use cancelable context for watchers

### PR DESCRIPTION
**Pull request by kramvan1**
Use a cancellable context for watchers

Fix gosec G601: Implicit memory aliasing in for loop for crawler.

### CODE REVIEW

File: rules/int_crawler.go [hunk 1]
```diff
@@ -166,7 +166,8 @@ func (ic *intCrawler) singleRun(logger *zap.Logger) {
 }
 func (ic *intCrawler) processData(values map[string]string, logger *zap.Logger) {
 	api := &cacheReadAPI{values: values}
-	for k, v := range values {
+	for k := range values {
+		v := values[k]
```
Comment (lcopi) [inline rules/int_crawler.go:170]: curious why this change?

**Review by lcopi — APPROVED** [review]

File: rules/int_crawler.go (hunk 1 above)
Comment (kramvan1) [reply rules/int_crawler.go:170]: `rules/int_crawler.go:174:22: G601: Implicit memory aliasing in for loop. (gosec)`

**Review by jswarren4 — APPROVED** [review]
